### PR TITLE
boards: arm: Add initial support for Renesas EK-RA2A1

### PR DESCRIPTION
**Pull request by soburi**
Add support for the Renesas EK-RA2A board.

This commit provides basic support only.
It just can run `blinky` and `button` samples.

### CODE REVIEW

**Comment by soburi** [comment]
It needs to rebase after #71017 is merged.

File: soc/renesas/ra/ra2a1/Kconfig
Comment (nordicjm) [inline soc/renesas/ra/ra2a1/Kconfig:35]: Should be dts properties

File: soc/renesas/ra/ra2a1/Kconfig
Comment (nordicjm) [inline soc/renesas/ra/ra2a1/Kconfig:14]: `if SOC_SERIES_RA2A1`

File: soc/renesas/ra/ra2a1/Kconfig
Comment (soburi) [reply soc/renesas/ra/ra2a1/Kconfig:35]: Thank you for reviewing.
This issue discussed in #71017.
I will apply the result of discussions.

**Comment by soburi** [comment]
Reimplementing on the FSP code base (#72150)

**Comment by soburi** [comment]
@thaoluonguw @KhiemNguyenT @ydamigos

File: drivers/serial/uart_renesas_ra8_sci.c
Comment (thenguyenyf) [inline drivers/serial/uart_renesas_ra8_sci.c:51]: k_spin_unlock should be called before return

File: dts/bindings/serial/renesas,ra8-uart-sci.yaml
Comment (thenguyenyf) [inline dts/bindings/serial/renesas,ra8-uart-sci.yaml:21]: These configurations are not used. It is better to remove it.

File: dts/arm/renesas/ra/ra2/ra2xx.dtsi
Comment (thenguyenyf) [inline dts/arm/renesas/ra/ra2/ra2xx.dtsi:231]: Do you refer to anywhere for this address value? As my understanding, it should be the address of OFS0 register as memtion in '7.2.1 Option Function Select Register 0 (OFS0) - RA2A1 MCU group HWM'

**Review by ydamigos — CHANGES_REQUESTED** [review]
Please update maintainers yaml to add the board.

File: west.yml
Comment (ydamigos) [inline west.yml:218]: HAL PR was merged. Please update it with hash.

File: drivers/serial/uart_renesas_ra8_sci.c
Comment (duynguyenxa) [inline drivers/serial/uart_renesas_ra8_sci.c:1]: @soburi , do you intend to update this file name as it's not target for RA8 device

**Review by ydamigos — DISMISSED** [review]

File: drivers/serial/uart_renesas_ra8_sci.c
Comment (soburi) [reply drivers/serial/uart_renesas_ra8_sci.c:1]: I cherry-picked your commit https://github.com/zephyrproject-rtos/zephyr/pull/76081/commits/b458878d6a16f8af4b549fd388d68ec301b6fc63 with few modifications.
Could you please check it?

File: dts/arm/renesas/ra/ra2/ra2xx.dtsi
Comment (soburi) [reply dts/arm/renesas/ra/ra2/ra2xx.dtsi]: FSP configures the options section.
So, this code can be removed, and I did so.

It may be necessary to accommodate additional settings, but this would require major changes, so we will consider this separately if necessary.

File: drivers/serial/uart_renesas_ra8_sci.c
Comment (soburi) [reply drivers/serial/uart_renesas_ra8_sci.c:1]: I was forget the https://github.com/zephyrproject-rtos/zephyr/commit/b458878d6a16f8af4b549fd388d68ec301b6fc63 SCI-UART driver requires FSP modification.

Your driver is more complete, so I would like to incorporate it.
Therefore, I would like to include it after merging https://github.com/zephyrproject-rtos/zephyr/pull/76081.

**Comment by soburi** [comment]
@duynguyenxa @ydamigos @KhiemNguyenT

https://github.com/zephyrproject-rtos/hal_renesas/pull/26

I applied this update. Could you re-review it, please?

File: soc/renesas/ra/ra2a1/Kconfig
Comment (nordicjm) [inline soc/renesas/ra/ra2a1/Kconfig:14]:
```suggestion
```

File: soc/renesas/ra/ra2a1/ram_sections.ld
Comment (nordicjm) [inline soc/renesas/ra/ra2a1/ram_sections.ld:10]: tab indent, not space

Comment (soburi) [reply soc/renesas/ra/ra2a1/ram_sections.ld]: Fixed it.

**Review by nordicjm — DISMISSED** [review]

**Comment by soburi** [comment]
@duynguyenxa @ydamigos @KhiemNguyenT

Could you take a look, please?

File: dts/arm/renesas/ra/ra2/r7fa2a1ab3cfm.dtsi
Comment (thaoluonguw) [inline dts/arm/renesas/ra/ra2/r7fa2a1ab3cfm.dtsi:11]: It is incorrect. Address should be "407e0000", could you please recheck?

Comment (soburi) [reply dts/arm/renesas/ra/ra2/r7fa2a1ab3cfm.dtsi]: https://www.renesas.com/us/en/document/mah/renesas-ra4m1-group-users-manual-hardware?r=1054146#G8.2206429

You're correct. Thank you for your pointing out.
Fixed it.

File: boards/renesas/ek_ra2a1/ek_ra2a1-pinctrl.dtsi
Comment (soburi) [reply boards/renesas/ek_ra2a1/ek_ra2a1-pinctrl.dtsi]: Fixed it.

File: boards/renesas/ek_ra2a1/ek_ra2a1-pinctrl.dtsi
Comment (soburi) [reply boards/renesas/ek_ra2a1/ek_ra2a1-pinctrl.dtsi]: Fixed it.

File: boards/renesas/ek_ra2a1/doc/index.rst
Comment (soburi) [reply boards/renesas/ek_ra2a1/doc/index.rst]: I applied all of your comments about the documentation.

I didn't include any photos or illustrations because I wasn't clear about the copyright permissions. If you can update, I hope you to do so.

File: west.yml
Comment (ydamigos) [inline west.yml:217]: Better use the latest hash.

**Review by ydamigos — CHANGES_REQUESTED** [review]

File: west.yml
Comment (soburi) [reply west.yml:217]: Update done.

**Comment by soburi** [comment]
@ydamigos 
Could you recheck it, please?

File: west.yml
Comment (ydamigos) [reply west.yml]: HAL was updated yesterday. Please update it one more.

**Comment by soburi** [comment]
@duynguyenxa @ydamigos @KhiemNguyenT @thaoluonguw 

Could you take a look again, please?

**Comment by soburi** [comment]
@ydamigos 

> > Could you take a look again, please?
> 
> HAL was updated yesterday. Please update west.yml once more.

Update done.

File: MAINTAINERS.yml
Comment (KhiemNguyenT) [inline MAINTAINERS.yml:3766]: Is it good to describe *ra* only ? Another PR is adding RA6 support, then RA4 will come.

Comment (soburi) [reply MAINTAINERS.yml]: That looks better your suggestion. I've updated it.

File: soc/renesas/ra/ra2a1/soc.c
Comment (soburi) [reply soc/renesas/ra/ra2a1/soc.c:40]: Fixed it.

File: boards/renesas/ek_ra2a1/ek_ra2a1_defconfig
Comment (soburi) [reply boards/renesas/ek_ra2a1/ek_ra2a1_defconfig:17]: Fixed it.

File: west.yml
Comment (soburi) [reply west.yml]: Update done.

File: soc/renesas/ra/ra2a1/soc.c
Comment (thenguyenyf) [reply soc/renesas/ra/ra2a1/soc.c:40]: Could you please re-check? I see it still exists.

**Review by ydamigos — APPROVED** [review]

File: boards/renesas/ek_ra2a1/doc/index.rst
Comment (thaoluonguw) [reply boards/renesas/ek_ra2a1/doc/index.rst]: I am sorry for my late. I have no comment for your PR. For the photos of RA2A1, we will update it in https://github.com/zephyrproject-rtos/zephyr/pull/77298 after this PR is merged.

**Review by thaoluonguw — APPROVED** [review]

**Comment by soburi** [comment]
@KhiemNguyenT 
Could you recheck and reapprove, please?

**Review by KhiemNguyenT — APPROVED** [review]

**Pull request merged**
